Cargo.toml

```diff
@@ -199,3 +199,15 @@ tests_outside_test_module = "deny"
 unwrap_in_result = "deny"
 unwrap_used = "deny"
 use_debug = "deny"
+
+[profile.release]
+codegen-units = 1
+lto = "fat" # attempts to perform optimizations across all crates within the dependency graph
+
+[profile.bench]
+inherits = "release"
+lto = "thin" # takes substantially less time to run than "fat" while still achieving similar performance gains
+
+[profile.profiling]
```

Comment:
I would just make this the bench profile, debug is only bigger binaries, you don't care about size in this case